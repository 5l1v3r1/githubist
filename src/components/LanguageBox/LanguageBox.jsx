// @flow

import React from 'react';
import { Link } from 'react-router-dom';

import styles from './LanguageBox.scss';

type Props = {
  name: string,
  slug: string,
  style: {
    backgroundColor: string,
    color: string,
  },
};

const LanguageBox = ({ name, slug, style }: Props) => (
  <Link to={`/language/${slug}`} className={styles.languageBox} style={style}>
    {name}
  </Link>
);

export default LanguageBox;
