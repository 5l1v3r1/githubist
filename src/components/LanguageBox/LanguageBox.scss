@import '../../theme/theme';

.languageBox {
  display: flex;
  width: 100%;
  height: px-to-rem(160px);
  align-items: center;
  justify-content: center;
  border-radius: px-to-rem(2px);
  font-size: map-get($heading-sizes, h3);
  font-weight: 600;
  text-decoration: none;
}
