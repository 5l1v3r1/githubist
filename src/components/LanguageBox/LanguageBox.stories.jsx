// @flow

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { storiesOf } from '@storybook/react';

import LanguageBox from './LanguageBox';

const colors = {
  javascript: {
    backgroundColor: '#FFC157',
    color: '#000',
  },

  java: {
    backgroundColor: '5781FF',
    color: '#fff',
  },
};

storiesOf('LanguageBox', module)
  .add('default', () => (
    <MemoryRouter>
      <div style={{ width: '250px' }}>
        <LanguageBox name="JavaScript" slug="javascript" style={colors.javascript} />
      </div>
    </MemoryRouter>
  ))
  .add('with light text', () => (
    <MemoryRouter>
      <div style={{ width: '250px' }}>
        <LanguageBox name="Java" slug="java" style={colors.java} />
      </div>
    </MemoryRouter>
  ));
