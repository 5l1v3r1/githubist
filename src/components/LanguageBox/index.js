// @flow

import LanguageBox from './LanguageBox';

export default LanguageBox;
