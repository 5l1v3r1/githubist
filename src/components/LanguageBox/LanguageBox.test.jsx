// @flow

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import LanguageBox from './LanguageBox';

const colors = {
  javascript: {
    backgroundColor: '#FFC157',
    color: '#000',
  },

  java: {
    backgroundColor: '5781FF',
    color: '#fff',
  },
};

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LanguageBox name="JavaScript" slug="javascript" style={colors.javascript} />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly light box', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LanguageBox name="Java" slug="java" style={colors.java} />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
