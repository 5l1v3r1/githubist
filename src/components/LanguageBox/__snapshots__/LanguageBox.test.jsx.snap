// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LanguageBox
      name="JavaScript"
      slug="javascript"
      style={
        Object {
          "backgroundColor": "#FFC157",
          "color": "#000",
        }
      }
    >
      <Link
        replace={false}
        style={
          Object {
            "backgroundColor": "#FFC157",
            "color": "#000",
          }
        }
        to="/language/javascript"
      >
        <a
          href="/language/javascript"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FFC157",
              "color": "#000",
            }
          }
        >
          JavaScript
        </a>
      </Link>
    </LanguageBox>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly light box 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LanguageBox
      name="Java"
      slug="java"
      style={
        Object {
          "backgroundColor": "5781FF",
          "color": "#fff",
        }
      }
    >
      <Link
        replace={false}
        style={
          Object {
            "backgroundColor": "5781FF",
            "color": "#fff",
          }
        }
        to="/language/java"
      >
        <a
          href="/language/java"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "5781FF",
              "color": "#fff",
            }
          }
        >
          Java
        </a>
      </Link>
    </LanguageBox>
  </Router>
</MemoryRouter>
`;
